src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IForEachLoopStatement.cs

```diff
@@ -788,8 +847,114 @@ static void Main()
                   ILocalReferenceExpression: num (OperationKind.LocalReferenceExpression, Type: System.Int32) (Syntax: 'num')
                   InConversion: null
                   OutConversion: null
+  NextVariables(0)
 ";
             VerifyOperationTreeForTest<ForEachStatementSyntax>(source, expectedOperationTree);
         }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact, WorkItem(17602, "https://github.com/dotnet/roslyn/issues/17602")]
+        public void IForEachLoopStatement_WithDeconstructDeclaration()
+        {
+            string source = @"
+class X
+{
+    public static void M((int, int)[] x)
+    {
+        /*<bind>*/foreach (var (a, b) in x)
+        {
+        }/*</bind>*/
+    }
+}
+";
+            string expectedOperationTree = @"
+IForEachLoopStatement (LoopKind.ForEach) (OperationKind.LoopStatement) (Syntax: 'foreach (va ... }')
+  Locals: Local_1: System.Int32 a
+    Local_2: System.Int32 b
+  LoopControlVariable: IOperation:  (OperationKind.None) (Syntax: 'var (a, b)')
```

Comment:
>LoopControlVariable: IOperation:  (OperationKind.None) (Syntax: 'var (a, b)') [](start = 2, length = 77)

I would expect to see references to locals ```a``` and ```b``` somewhere in this sub-tree.

Reply:
Filed https://github.com/dotnet/roslyn/issues/22032